### ISSUE TITLE
[MaterialDatePicker] Strange behaviour of choosing year on "weak" device

### ISSUE THREAD

**Issue by achatina** [material-components/material-components-android#707]
**Description:** 
I have implemented MaterialDatePicker in my application, but when I'm trying to change the year on my Redmi 4X, it starts lagging and sometimes it stops on the wrong year. On the gif below I have selected 2015.

![material_date_picker](https://user-images.githubusercontent.com/16122829/67492642-63f83080-f67f-11e9-92b6-44d3d4424e4e.gif)

**Expected behavior:** It selects proper year without lags

**Source code:** 
```
val constraintsBuilder = CalendarConstraints.Builder()
val builder = MaterialDatePicker.Builder.dateRangePicker()
builder.setCalendarConstraints(constraintsBuilder.build())
builder.setTheme(resolveOrThrow(context!!, R.attr.materialCalendarTheme))
val picker = builder.build()
picker.addOnPositiveButtonClickListener {
            dateSelected(field, createDateRange(it.first ?: 0, it.second ?: 0), value)
}
picker.show(fragmentManager!!, picker.toString())
```

**Android API version:** Android 7.1.2, API 25

**Material Library version:** 
1.2.0-alpha01
1.1.0-beta01
(haven't tested on the 1.1.0-alpha versions

**Device:** Xiaomi Redmi 4X 3/32

**Comment by achatina** [comment]
*On emulator everything works much better

**Comment by achatina** [comment]
I have tested this on your demo catalog app. The bug is here too. 

![device-2019-10-24-203222](https://user-images.githubusercontent.com/16122829/67510455-a892c480-f69d-11e9-9faa-20c97c1b3a90.gif)

**Comment by ldjcmu** [comment]
Fixed with a5ec8c66431faba6c601f54071cfbcc13b961f3f
